api/src/main/java/org/apache/iceberg/catalog/Namespace.java

```diff
@@ -29,6 +31,8 @@
 public class Namespace {
   private static final Namespace EMPTY_NAMESPACE = new Namespace(new String[] {});
   private static final Joiner DOT = Joiner.on('.');
+  private static final Predicate<String> CONTAINS_NULL_BYTE =
```

Comment:
`BYTE` probably isn't correct since this is checking for the null unicode codepoint. Maybe just `CONTAINS_NULL`?

Reply:
That's fair. Maybe just `CONTAINS_NULL_CHARACTER`? `CONTAINS_NULL` makes it sound like we're checking for `null` itself, which this regex does not do.